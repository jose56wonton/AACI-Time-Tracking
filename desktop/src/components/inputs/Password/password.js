import React from 'react';
import PropTypes from 'prop-types';

import { TextField as MUTextField } from '@material-ui/core';
import { getIn } from 'formik';

const TextField = ({ field, form, id, label, helper,margin }) => (
  <MUTextField
    {...field}    
    id={id}
    label={label}
    type="password"
    InputLabelProps={{ shrink: true }}
    fullWidth
    margin={margin}
    FormHelperTextProps={helper ==='none' ? { style:{ display:'none' },error: true } : { error: true  }}
    helperText={getIn(form.errors, field.name)|| ' '} 
  />
);

TextField.defaultProps = {  
  helper: 'normal',
  margin: 'normal'
};

TextField.propTypes = {
  field: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  id: PropTypes.string,
  label: PropTypes.string.isRequired,
  helper: PropTypes.oneOf(['none','normal'])
};

export default TextField;
