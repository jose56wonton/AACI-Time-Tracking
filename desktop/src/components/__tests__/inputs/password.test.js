import React from 'react';
import { shallow } from 'enzyme';

import { Password } from 'components/inputs/Password/password';
import PasswordHOC from 'components/inputs/Password';

const props =  {  
  form:{
    errors: {}
  },
  field: {
    name: 'name',
  },
  label: 'label',
  labelProps: {},
  formControlProps: {},
  margin: 'normal',
  classes: {},
  helper: 'normal',
  fullWidth: true
};

const setup = overRides => {  
  return shallow(<Password {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return shallow(<PasswordHOC {...props} {...overRides}/>);
};

describe('Password Input', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();   
  }); 
});