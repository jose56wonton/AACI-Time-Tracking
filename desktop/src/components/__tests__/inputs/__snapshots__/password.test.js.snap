// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Input should render correctly 1`] = `
<TextField
  FormHelperTextProps={
    Object {
      "error": true,
    }
  }
  InputLabelProps={
    Object {
      "shrink": true,
    }
  }
  fullWidth={true}
  helperText=" "
  label="label"
  margin="normal"
  name="name"
  required={false}
  select={false}
  type="password"
  variant="standard"
/>
`;

exports[`Password Input should render correctly withStyles 1`] = `
<TextField
  FormHelperTextProps={
    Object {
      "error": true,
    }
  }
  InputLabelProps={
    Object {
      "shrink": true,
    }
  }
  fullWidth={true}
  helperText=" "
  label="label"
  margin="normal"
  name="name"
  required={false}
  select={false}
  type="password"
  variant="standard"
/>
`;
